didcomm_resolver/resolver.py

```diff
@@ -98,7 +92,32 @@ async def update_connection(
         methods: Sequence[str],
     ):
         """Update resolvers supported methods."""
-        return await cls.register_connection(session, connection_id, methods)
+        return await cls._upsert_methods(session, connection_id, methods, update=True)
+
+    @classmethod
+    async def _upsert_methods(cls, session, connection_id, methods, update=False):
+
+        # Remove possible duplicates methods in the request
+        methods = list(dict.fromkeys(methods))
```

Comment:
another way could be, `[*{*methods}]`, but `list(set(methods))` is faster and more readable. We should also be able to represent methods as a set everywhere, removing the need to cast to a list.